src/Service.Tests/Configuration/LoadConfigViaEndpoint.cs

```diff
@@ -0,0 +1,98 @@
+    {
+        TestServer server = new(Program.CreateWebHostFromInMemoryUpdateableConfBuilder(Array.Empty<string>()));
+        HttpClient httpClient = server.CreateClient();
+
+        RuntimeConfig config = ReadCosmosConfigurationFromFile() with { Schema = "@env('schema')" };
+
+        ConfigurationPostParametersV2 @params = new(
+            Configuration: config.ToJson(),
+            ConfigurationOverrides: "{}",
+            Schema: @"
+            type Entity {
+                id: ID!
+                name: String!
+            }
+            ",
+            AccessToken: null
+        );
+
+        JsonContent content = JsonContent.Create(@params);
+
+        HttpResponseMessage postResult =
+            await httpClient.PostAsync(configurationEndpoint, content);
+        Assert.AreEqual(HttpStatusCode.OK, postResult.StatusCode);
+
+        RuntimeConfigProvider configProvider = server.Services.GetService(typeof(RuntimeConfigProvider)) as RuntimeConfigProvider;
+        RuntimeConfig loadedConfig = configProvider.GetConfig();
+
+        Assert.AreEqual(config.Schema, loadedConfig.Schema);
+    }
+
+    [TestMethod("Testing that environment variables can be replaced at runtime not only when config is loaded."), TestCategory(TestCategory.COSMOSDBNOSQL)]
+    [DataRow(CONFIGURATION_ENDPOINT)]
+    [DataRow(CONFIGURATION_ENDPOINT_V2)]
+    [Ignore("We don't want to environment variable substitution in late configuration, but test is left in for if this changes.")]
+    public async Task CanLoadConfigWithEnvironmentVariables(string configurationEndpoint)
+    {
+        Environment.SetEnvironmentVariable("schema", "schema.graphql");
+        TestServer server = new(Program.CreateWebHostFromInMemoryUpdateableConfBuilder(Array.Empty<string>()));
+        HttpClient httpClient = server.CreateClient();
+
+        RuntimeConfig config = ReadCosmosConfigurationFromFile() with { Schema = "@env('schema')" };
+
+        ConfigurationPostParametersV2 @params = new(
+            Configuration: config.ToJson(),
+            ConfigurationOverrides: "{}",
+            Schema: @"
+            type Entity {
+                id: ID!
+                name: String!
+            }
+            ",
+            AccessToken: null
+        );
+
+        JsonContent content = JsonContent.Create(@params);
+
+        HttpResponseMessage postResult =
+            await httpClient.PostAsync(configurationEndpoint, content);
+        Assert.AreEqual(HttpStatusCode.OK, postResult.StatusCode);
+
+        RuntimeConfigProvider configProvider = server.Services.GetService(typeof(RuntimeConfigProvider)) as RuntimeConfigProvider;
+        RuntimeConfig loadedConfig = configProvider.GetConfig();
+
+        Assert.AreNotEqual(config.Schema, loadedConfig.Schema);
+        Assert.AreEqual(Environment.GetEnvironmentVariable("schema"), loadedConfig.Schema);
+    }
+
+    [TestCleanup]
+    public void Cleanup()
+    {
+        Environment.SetEnvironmentVariable("schema", null);
```

Comment:
Why clean if the env var is not there in the first place?

Reply:
I see it is in the ignored test. So i guess this doesn't hurt.

Reply:
True, it doesn't _need_ to be done since the test is ignored, but if we didn't have it, chances are we'd add the test back in but not add that in and it'd result in unexpected side effects.